wit/request-reply.wit

```diff
@@ -0,0 +1,16 @@
+/// The request-reply interface allows a guest to send a message and await a response. This
+/// interface is considered optional as not all message services support the concept of
+/// request/reply. However, request/reply is a very common pattern in messaging and as such, we have
+/// included it as a core interface.
+interface request-reply {
+    use types.{client, message, error};
+
+    /// Performs a blocking request/reply operation with an optional timeout. If the timeout value
+    /// is not set, then the request/reply operation will block indefinitely.
+    request: func(c: client, msg: message, timeout-ms: option<u32>) -> result<option<list<message>>, error>;
```

Comment:
```suggestion
    request: func(c: client, msg: message, options: option<request-options>) -> result<option<list<message>>, error>;
```

```wit
record request-options {
  timeout-ms: option<u32>
}
```

I suggest avoiding `option` as the function's signature to avoid breaking changes from the caller and instead use an object as a third parameter

Reply:
With the inclusion of the new `@since` stuff, this is probably the way to go. I'll make the change